core/impl-base/src/main/java/org/jboss/arquillian/core/impl/ManagerImpl.java

```diff
@@ -597,6 +599,27 @@ void debug(ObserverMethod method, boolean interceptor)
       }
    }
 
+   private void debugExtension(Class<?> extension)
```

Comment:
I would also consider refactoring this logging logic to separate component, this class is growing and holds too many resp. Would you mind doing it as part of this PR?